### ISSUE TITLE
Cannot fetch from "W3C PR-math-19980224"

### ISSUE THREAD

**Issue by manuelfuenmayor** [relaton/relaton-w3c#38]
I'm trying to auto-fetch the reference of this document: https://www.w3.org/TR/1998/PR-math-19980224/,
with no success:

```
$ bundle exec relaton fetch "W3C PR-math-19980224"
[relaton-w3c] (W3C PR-math-19980224) Fetching from Relaton repository ...
[relaton-w3c] Downloaded index from https://raw.githubusercontent.com/relaton/relaton-data-w3c/main/index1.zip
[relaton-w3c] (W3C PR-math-19980224) Not found.
No matching bibliographic entry found

```

**Comment by ronaldtse** [comment]
I've also tried these:

```
[relaton-w3c] (W3C TR PR-math-19980224) Fetching from Relaton repository ...
[relaton-w3c] (W3C TR PR-math-19980224) Not found.
No matching bibliographic entry found
[relaton-w3c] (W3C TR math-19980224) Fetching from Relaton repository ...
[relaton-w3c] (W3C TR math-19980224) Not found.
No matching bibliographic entry found
```

**Comment by andrew2net** [comment]
@ronaldtse the isn't `W3C PR-math-19980224` document in the [dataset](http://www.w3.org/2002/01/tr-automation/tr.rdf). There is only one relation which obsolete the document. Do we need obsoleted docs in our data repo?
```xml
<REC rdf:about="https://www.w3.org/TR/1998/REC-MathML-19980407/">
  <dc:date>1998-04-07</dc:date>
  <dc:title>Mathematical Markup Language (MathML) 1.0 Specification</dc:title>
  <doc:obsoletes rdf:resource="https://www.w3.org/TR/1998/PR-math-19980224"/>
  <doc:versionOf rdf:resource="https://www.w3.org/TR/REC-MathML/"/>
  <editor rdf:parseType="Resource">
    <contact:fullName>Patrick D F Ion</contact:fullName>
  </editor>
  <editor rdf:parseType="Resource">
    <contact:fullName>Robert R Miner</contact:fullName>
  </editor>
  <org:deliveredBy rdf:parseType="Resource">
    <contact:homePage rdf:resource="https://www.w3.org/Math/"/>
  </org:deliveredBy>
  <mat:hasErrata rdf:resource="https://www.w3.org/MarkUp/mathml101-updates/errata.html"/>
</REC>
```

**Comment by ronaldtse** [comment]
Yes we need all documents included obsoleted ones in the dataset. Being obsolete means others still cite it.

**Comment by andrew2net** [comment]
Then we need to scrape documents missed in the tr.rdf from www.w3.org website. We don't know all the missed documents. We can check if a relation is missed in tr.rdf. If it is, get it from www.w3.org.

**Comment by andrew2net** [comment]
@ronaldtse I see what happend. The most recent `tr.rdf` file doesn't have obsoleted docs. So we created an archive repo https://github.com/relaton/w3c-tr-archive. But the repo hasn't been updated for two years. Where can we get the previous `tr.rdf` files?

**Comment by ronaldtse** [comment]
@andrew2net yes you are correct. That's likely the only way we can get the full archive.

**Comment by andrew2net** [comment]
@ronaldtse wouldn't it be better to scrape all W3C documents from https://www.w3.org/TR/?
Documents have history
<img width="1011" alt="image" src="https://github.com/user-attachments/assets/70f45314-1d16-4403-a054-91010ed134e2">

In history, we can find obsoleted docs
https://www.w3.org/standards/history/REC-MathML/
<img width="839" alt="image" src="https://github.com/user-attachments/assets/963cbfd5-bc2d-4873-8f91-69a97945cbe3">

**Comment by ronaldtse** [comment]
@andrew2net then let's scrape that, but I don't think the details are as complete as the RDF file. So we have to combine?

**Comment by andrew2net** [comment]
@ronaldtse you are right. RDF data has more details.

We need to do the following:
- Download all archives.
- Because each archive file mostly repeats the previous version and passing all of them takes too much time, merge all the files into one `w3c.rdf` file, replacing old document records with newer ones.
- Update the W3C data fetcher to merge current `tr.rdf` into the `w3c.rdf` daly.

Once we do it we will have as much as we can from the archives. The `w3c.rdf` will only miss documents obsoleted before the first available archive was created. So documents that obsolete the missing docs will be in the `w3c.rdf`. That means we can check the presence of all `obsoletes` relations in the `w3c.rdf` and fetch it from W3C website if it doesn't exist.

**Comment by andrew2net** [comment]
@ronaldtse I fetched all the `tr.rdf` archives from web.archive.org and merged them into one archive. There are 9k+ documents in the merged file (we had 6k+ before). But the `W3C PR-math-19980224` still does not present. So we need to get such docs from W3C website.

**Comment by andrew2net** [comment]
@ronaldtse I'm trying to get all obsoleted relation that missed in the RDF archive from w3c.org website but the pages have inconsistent layout. I adapted scraper to get data from some layouts but it looks like there are much more of them. And some layouts conflict with each other. I don't know how much time will it take to cover all the variants, and want to clarify if I should continue to invest time to this problem.

**Comment by andrew2net** [comment]
This issue should be resolved by the switch to the official W3C API for data fetching (#44, commit cd9503d).

`DataFetcher` now iterates each specification's `version_history` via api.w3.org, which exposes obsoleted/historical versions including `PR-math-19980224`. Verified end-to-end:

- API: `GET https://api.w3.org/specifications/REC-MathML/versions/19980224` returns the Proposed Recommendation from 1998-02-24.
- Data repo: [`data/pr-math-19980224.yaml`](https://github.com/relaton/relaton-data-w3c/blob/main/data/pr-math-19980224.yaml) is now published with full metadata and relations (`editionOf REC-MathML`, `obsoletes WD-math-980106`, `updatedBy REC-MathML-19980407`).
- Index: `index1.yaml` contains the `{code: math, stage: PR, date: 19980224}` entry, which matches `PubId.parse("PR-math-19980224")`, so `relaton fetch "W3C PR-math-19980224"` now resolves.

Closing as resolved.
